File: mobile/android/wear/src/main/kotlin/com/khunor/lifey/ui/theme/LifeyColors.kt
package com.khunor.lifey.ui.theme

import androidx.compose.ui.graphics.Color

/**
 * The Lifey brand's dark-only color palette (41-watch-design-prompt.md §2),
 * as flat constants — this is the single place the raw hex values live so
 * every screen references the same tokens instead of re-declaring private
 * per-file `Color(0x...)` vals (docs/40-watch-app-plan.md §12.1 B6).
 *
 * `heart` deviates from the prompt's own §2.4 table (`#C46A6A`): every single
 * frame in the shipped design canvas (`docs/watch/design/Lifey Watch
 * Design.dc.html`) — Apple Watch, Wear OS, and the phone screens alike —
 * uses `#D97F7F` for the heart-rate icon/number instead. Same tie-break rule
 * the 42-doc's D0.1 applied to the elapsed-time color (canvas over prompt,
 * since the canvas is the later, visually-checked artifact): this file
 * follows the canvas. Flagged here in case that was an unintentional drift
 * rather than a deliberate re-pick.
 */
object LifeyColors {
    // 2.1 Surfaces
    /** True AMOLED black — what the screen itself sits on, not a token from
     * §2.1, but explicitly sanctioned by §2.1's own note ("on watch: may sit
     * on #000000"). */
    val trueBlack = Color(0xFF000000)
    val bg = Color(0xFF161611)
    val surface = Color(0xFF1C1E16)
    val container = Color(0xFF22241B)
    val containerHigh = Color(0xFF2A2C20)
    val containerHighest = Color(0xFF32342A)
    val outline = Color(0xFF3C3E32)

    // 2.2 Accents
    val primary = Color(0xFF9DAE6B)
    val secondary = Color(0xFFC49A6C)
    val tertiary = Color(0xFF6E9A6A)
    val primaryContainer = Color(0xFF22241B)
    val secondaryContainer = Color(0xFF2A2018)
    val tertiaryContainer = Color(0xFF1A2E1A)

    // 2.3 Text
    val onSurface = Color(0xFFF1F0E4)
    val onSurfaceVariant = Color(0xFFA8A899)
    val onPrimary = Color(0xFF161611)

    // 2.4 Metric accents — see class doc for the `heart` canvas/prompt note
    val heart = Color(0xFFD97F7F)
    val calories = Color(0xFFE0915A)
    val positive = Color(0xFF9DAE6B)
    val negative = Color(0xFFE08A52)

    // 2.5 Error
    val error = Color(0xFFCF6679)
    val onError = Color(0xFF1C0008)
    val errorContainer = Color(0xFF8C1D2F)
    val onErrorContainer = Color(0xFFFFB3BF)

    /** Dimmed content color for a ghosted/inert control — the log-set
     * control's pending and phone-unreachable states (canvas W 08/W 10,
     * docs/watch/43-watch-f5-set-logging-plan.md §3.2/§4.4). Mirrors iOS's
     * `LifeyColors.ghostedOnSurface`, added there first (S8) with a note to
     * add the identical hex here once Wear's own log-lap (S13) needed it. */
    val ghostedOnSurface = Color(0xFF46463E)

    /** The `phonelink_off` header glyph's quiet tint (docs/watch/
     * 44-watch-f6-standalone-plan.md §3.4, canvas W 13's "mode, not alarm" —
     * no chip/background of its own). Mirrors iOS's `LifeyColors
     * .standaloneIndicator`, added there first (S11). */
    val standaloneIndicator = Color(0xFF777264)

    // 2.6 Cardio activity-type accents (docs/cardio/55-cardio-watch-plan.md §2,
    // picker rows + active screens — C5.6) — the same hex the mobile app's
    // `MetricColors` uses for the matching activity, mirrors iOS's identical
    // "Cardio activity-type accents" LifeyColors section. `RUNNING` reuses
    // `calories` (`0xE0915A` on both), `HIKING` reuses `tertiary`
    // (`0x6E9A6A`), and `CYCLING` (docs/cardio/62-cardio-cycling-plan.md A5)
    // reuses `secondary` (`0xC49A6C`) — the mobile app's `colorScheme
    // .secondary`, for the same "every AppMetricColors slot already claimed"
    // reason documented there — rather than duplicating any of these three
    // hexes under a second name. Only the four colors this object didn't
    // already have get their own constant.
    /** `WALKING` — mirrors mobile's `MetricColors.steps`. */
    val cardioWalking = Color(0xFFB08AC8)
    /** `INDOOR_BIKE` — mirrors mobile's `MetricColors.carbs`. */
    val cardioIndoorBike = Color(0xFFD8B35A)
    /** `BASKETBALL` — mirrors mobile's `MetricColors.fat`. */
    val cardioBasketball = Color(0xFF8E8EC4)
    /** `FOOTBALL` — mirrors mobile's `MetricColors.water`. */
    val cardioFootball = Color(0xFF6FA8C4)
}

File: mobile/ios/LifeyWatch/Theme/LifeyColors.swift
import SwiftUI

extension Color {
  init(hex: UInt32) {
    self.init(
      red: Double((hex >> 16) & 0xFF) / 255,
      green: Double((hex >> 8) & 0xFF) / 255,
      blue: Double(hex & 0xFF) / 255)
  }
}

/// The Lifey brand's dark-only color palette (41-watch-design-prompt.md §2),
/// as flat constants — mirrors Android's `LifeyColors.kt` value-for-value
/// (docs/40-watch-app-plan.md §12.1 B6) so both watch platforms render the
/// same hex, not just the same token names.
///
/// `heart` deviates from the prompt's own §2.4 table (`#C46A6A`): every
/// single frame in the shipped design canvas (`docs/watch/design/Lifey Watch
/// Design.dc.html`) — Apple Watch, Wear OS, and the phone screens alike —
/// uses `#D97F7F` for the heart-rate icon/number instead. Same tie-break
/// rule the 42-doc's D0.1 applied to the elapsed-time color (canvas over
/// prompt, since the canvas is the later, visually-checked artifact): this
/// file follows the canvas, matching Android's own `LifeyColors.kt` note.
enum LifeyColors {
  // 2.1 Surfaces
  /// True AMOLED black — what the screen itself sits on, not a token from
  /// §2.1, but explicitly sanctioned by §2.1's own note ("on watch: may sit
  /// on #000000").
  static let trueBlack = Color(hex: 0x00_00_00)
  static let bg = Color(hex: 0x16_16_11)
  static let surface = Color(hex: 0x1C_1E_16)
  static let container = Color(hex: 0x22_24_1B)
  static let containerHigh = Color(hex: 0x2A_2C_20)
  static let containerHighest = Color(hex: 0x32_34_2A)
  static let outline = Color(hex: 0x3C_3E_32)

  // 2.2 Accents
  static let primary = Color(hex: 0x9D_AE_6B)
  static let secondary = Color(hex: 0xC4_9A_6C)
  static let tertiary = Color(hex: 0x6E_9A_6A)

  // 2.3 Text
  static let onSurface = Color(hex: 0xF1_F0_E4)
  static let onSurfaceVariant = Color(hex: 0xA8_A8_99)
  static let onPrimary = Color(hex: 0x16_16_11)

  // 2.4 Metric accents — see the enum doc for the `heart` canvas/prompt note
  static let heart = Color(hex: 0xD9_7F_7F)
  static let calories = Color(hex: 0xE0_91_5A)
  static let positive = Color(hex: 0x9D_AE_6B)
  static let negative = Color(hex: 0xE0_8A_52)

  // 2.5 Error — added for the F5 log-set "failed" toast (docs/watch/
  // 43-watch-f5-set-logging-plan.md §3.2); mirrors Android's `LifeyColors.kt`,
  // which already has this section.
  static let error = Color(hex: 0xCF_66_79)
  static let onError = Color(hex: 0x1C_00_08)
  static let errorContainer = Color(hex: 0x8C_1D_2F)
  static let onErrorContainer = Color(hex: 0xFF_B3_BF)

  /// Dimmed content color for a ghosted/inert control — the log-set
  /// control's pending and phone-unreachable states (canvas AW 09/AW 11,
  /// docs/watch/43-watch-f5-set-logging-plan.md §3.2/§4.4). Not yet in
  /// Android's `LifeyColors.kt` since Wear's own log-lap ships later (S13)
  /// — add the identical hex there when it does.
  static let ghostedOnSurface = Color(hex: 0x46_46_3E)

  /// The standalone mode indicator's own muted tone — canvas AW 14/W 13
  /// explicitly calls for something quieter than `onSurfaceVariant`
  /// (docs/watch/44-watch-f6-standalone-plan.md §3.4: "mode, not alarm —
  /// muted `777264`"), since it's a passive glyph next to the header label,
  /// not informational text. Not yet in Android's `LifeyColors.kt` since
  /// Wear's own active-screen delta ships later (S17) — add the identical
  /// hex there when it does.
  static let standaloneIndicator = Color(hex: 0x77_72_64)

  // 2.6 Cardio activity-type accents (docs/cardio/55-cardio-watch-plan.md §2,
  // picker rows — C5.4) — the same hex the mobile app's `MetricColors` uses
  // for the matching activity (`activity_type.dart`'s `activityTypeColor`),
  // kept identical across platforms like every other token in this file.
  // `RUNNING` reuses `calories` (`0xE0915A` on both), `HIKING` reuses
  // `tertiary` (`0x6E9A6A`), and `CYCLING` (docs/cardio/62-cardio-cycling-
  // plan.md A6) reuses `secondary` (`0xC49A6C`) — the mobile app's
  // `colorScheme.secondary`, for the same "every AppMetricColors slot
  // already claimed" reason documented there — rather than duplicating any
  // of these three hexes under a second name. Only the four colors this
  // file didn't already have get their own constant.
  /// `WALKING` — mirrors mobile's `MetricColors.steps`.
  static let cardioWalking = Color(hex: 0xB0_8A_C8)
  /// `INDOOR_BIKE` — mirrors mobile's `MetricColors.carbs`.
  static let cardioIndoorBike = Color(hex: 0xD8_B3_5A)
  /// `BASKETBALL` — mirrors mobile's `MetricColors.fat`.
  static let cardioBasketball = Color(hex: 0x8E_8E_C4)
  /// `FOOTBALL` — mirrors mobile's `MetricColors.water`.
  static let cardioFootball = Color(hex: 0x6F_A8_C4)
}
